base/math.jl

```diff
@@ -411,25 +411,25 @@ Return `(x,exp)` such that `x` has a magnitude in the interval ``[1/2, 1)`` or 0
 and `val` is equal to ``x \\times 2^{exp}``.
 """
 function frexp{T<:AbstractFloat}(x::T)
-    xu = reinterpret(Unsigned,x)
+    xu = reinterpret(Unsigned, x)
     xe = xu & exponent_mask(T)
+    xe == exponent_mask(T) && return x, 0 # NaN or Inf
     k = Int(xe >> significand_bits(T))
     if xe == 0 # x is subnormal
-        x == 0 && return x, 0
         xs = xu & sign_mask(T)
         xu $= xs
-        m = leading_zeros(xu)-exponent_bits(T)
+        xu == 0 && return x, 0 # +-0
```

Comment:
Since we already have the  bit representation, it's better to   check the zero here as opposed to `x == 0`